YCDOCIO-1617: Terraform for ALB Ingress Controller

vitaly-belyakov opened the pull request:
added TF-manifest for Kubernetes cluster with ALB Ingress Controller

vitaly-belyakov committed:
YCDOCIO-1607: added TF manifest for standard k8s cluster

vitaly-belyakov committed:
YCDOCIO-1607: added rules for security group (443 & 6443 ports)

vitaly-belyakov committed:
YCDOCIO-1607: added code description

vitaly-belyakov committed:
YCDOCIO-1607: edits in TF manifest and code description

vitaly-belyakov committed:
YCDOCIO-1607: deleted code description

vitaly-belyakov committed:
YCDOCIO-1607: edit after review

vitaly-belyakov committed:
YCDOCIO-1607: edit after review

vitaly-belyakov committed:
YCDOCIO-1617: added TF-manifest for Kubernetes cluster with ALB Ingress Controller

vitaly-belyakov committed:
Delete k8s-cluster.tf

vitaly-belyakov committed:
YCDOCIO-1617: edit after review

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,175 @@
+#
+# Set the configuration of Managed Service for Kubernetes cluster
+
+locals {
```
Точно. Неформатированность конфига считаем грубой ошибкой.

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,175 @@
+  k8s_cluster_version = "1.21" # Set the version of Kubernetes for the master host
+  zone_a_v4_cidr_blocks = "10.1.0.0/16" # Set the CIDR block for the Subnet in ru-central1-a availability zone
+}
+
```
Не стоит. На созвоне было решено, что комментарии нужны только для ресурсов, у которых нет поля `description`. Пока не поступило других указаний, действуем как сказали.

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,175 @@
+  name        = "k8s-network"
+  description = "Network for the Managed Service for Kubernetes cluster"
+}
+
```
Достаточно просто переместить `description` в начало описания ресурса. Комментарий не нужен.

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,177 @@
+
+  scale_policy {
+    fixed_scale {
+      size = 1 # Number of hosts
```
Count?

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,177 @@
+  description    = "Subnet in ru-central1-a availability zone"
+  zone           = "yandex_vpc_subnet.subnet-a.zone"
+  network_id     = yandex_vpc_network.k8s-network.id
+  v4_cidr_blocks = ["10.1.0.0/16"]
```
Так и просится перехать в `locals`.

vitaly-belyakov replied:
точнее, в vars. В locals почему-то не взлетает

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,177 @@
+  ingress {
+    protocol       = "TCP"
+    description    = "The rule allows availability checks from the load balancer's range of addresses. It is required for the operation of a fault-tolerant cluster and load balancer services."
+    v4_cidr_blocks = ["198.18.235.0/24", "198.18.248.0/24"] # The load balancer's address range
```
Эти адреса оба хотят в `locals`. Если, конечно, для них не справедливо вот это:

> `predefined_target` (Optional) - Special-purpose targets. `self_security_group` refers to this particular security group. `loadbalancer_healthchecks` represents [loadbalancer health check nodes](https://cloud.yandex.com/docs/network-load-balancer/concepts/health-check).

Взял отсюда: https://github.com/yandex-cloud/terraform-provider-yandex/blob/master/website/docs/r/vpc_security_group.html.markdown#argument-reference

maxdunaevsky-documentatio replied:
@vitaly-belyakov , привет!

Ты в трекере написал, что сработало, а тут ничего не исправил. Сделай правки, пожалуйста, здесь и в 46-ой.

vitaly-belyakov committed:
YCDOCIO-1617: edit after review

vitaly-belyakov committed:
YCDOCIO-1617: updated load balancer rule and other edits

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,179 @@
+
+  ingress {
+    protocol       = "TCP"
+    description    = "The rule allows incoming TCP-connections on ports 10501 and 10502 from the load balancer's range of addresses."
```
Поле `description` должно идти первым здесь и в остальных ресурсах.

maxdunaevsky-documentatio reviewed (changes requested):
Исправлены не все замечания.

vitaly-belyakov committed:
YCDOCIO-1617: edit after review

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,179 @@
+# Infrastructure for Yandex Cloud Managed Service for Kubernetes cluster and Application Load Balancer Ingress Controller.
+#
```
Не хватает ссылок на статьи. Посмотри, как в соседних файлах сделано. Нужно добавить такое:

```terraform
# EN: https://cloud.yandex.com/en/docs/managed-kubernetes/tutorials/alb-ingress-controller
# RU: https://cloud.yandex.ru/docs/managed-kubernetes/tutorials/alb-ingress-controller
```

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,179 @@
+  zone_a_v4_cidr_blocks  = "10.1.0.0/16" # Set the CIDR block for the Subnet in ru-central1-a availability zone.
+}
+
+variable "zone_a_v4_cidr_blocks" {
```
У нас дома уже есть `zone_a_v4_cidr_blocks`.

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,179 @@
+  folder_id              = ""            # Set your cloud folder ID.
+  k8s_node_group_version = "1.20"        # Set the version of Kubernetes for the node group.
+  k8s_cluster_version    = "1.21"        # Set the version of Kubernetes for the master host.
+  zone_a_v4_cidr_blocks  = "10.1.0.0/16" # Set the CIDR block for the Subnet in ru-central1-a availability zone.
```
`zone_a_v4_cidr_blocks` дома.

maxdunaevsky-documentatio commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,179 @@
+}
+
+resource "yandex_vpc_network" "k8s-network" {
+  description = "Network for the Managed Service for Kubernetes cluster."
```
1. Сократи все `description` до одного предложения и убери точки в конце.
2. Точки в конце комментариев оставь.

maxdunaevsky-documentatio reviewed (changes requested):
1. `zone_a_v4_cidr_blocks` дублируется в `locals` и `variable`.  There can be only one (`locals`).
2. Сократи `description` до одного предложения, убери точки в конце.
3. В конце комментариев ничего не делай, оставь точки, с ними всё хорошо.

vitaly-belyakov committed:
YCDOCIO-1617: edit after review

vitaly-belyakov commented:
> 1. `zone_a_v4_cidr_blocks` дублируется в `locals` и `variable`.  There can be only one (`locals`).

в locals оно почему-то не работает, только в vars

vitaly-belyakov commented on tutorials/terraform/managed-kubernetes/k8s-alb-ic.tf:
```diff
@@ -0,0 +1,177 @@
+
+  allocation_policy {
+    location {
+      zone = "yandex_vpc_subnet.subnet-a.zone"
```
без кавычек работает, проверил

vitaly-belyakov committed:
YCDOCIO-1617: edit after review

Commit:
Merge branch 'master' into YCDOCIO-1617-TF-manifest-ALB-IC

Commit:
Some arguments moved to locals

Commit:
Rework roles, ports and other

vitaly-belyakov committed:
YCDOCIO-1617: added security group to node group

vitaly-belyakov committed:
added dot

vitaly-belyakov committed:
YCDOCIO-1617: small fixes

vitaly-belyakov committed:
Merge branch 'YCDOCIO-1617-TF-manifest-ALB-IC' of https://github.com/documentat/yandex-cloud-examples into YCDOCIO-1617-TF-manifest-ALB-IC

voloschuk-fedor committed:
Update k8s-alb-ingress-controller.tf

- Убрал правила для портов 10501 и 10502, они не нужны.
- Добавил правило для IP-адресов хелсчеков. Без этого правила балансировщик не запускается.
- Сделал одну версию для мастера и группы узлов.

AnatolyKraev committed:
Merge branch 'master' into YCDOCIO-1617-TF-manifest-ALB-IC

AnatolyKraev committed:
YCDOCIO-1617: remove version k8s

AnatolyKraev committed:
YCDOCIO-1617: fix k8s version

AnatolyKraev committed:
YCDOCIO-1617: return k8s version

AnatolyKraev committed:
YCDOCIO-1617: return k8s version

AnatolyKraev committed:
YCDOCIO-1617: return k8s version
